packages/core/entity/FileDetail/index.ts

```diff
@@ -160,24 +161,36 @@ export default class FileDetail {
         return this.fileDetail.annotations.find((annotation) => annotation.name === annotationName);
     }
 
-    public getPathToThumbnail(): string | undefined {
+    public async getPathToThumbnail(): Promise<string | undefined> {
+        // If the thumbnail is a relative path on the allen drive then preprend it to
+        // the AICS FMS NGINX server path
+        if (this.thumbnail?.startsWith("/allen")) {
+            return `${AICS_FMS_FILES_NGINX_SERVER}${this.thumbnail}`;
+        }
+
         // If no thumbnail present try to render the file itself as the thumbnail
         if (!this.thumbnail) {
             const isFileRenderableImage = RENDERABLE_IMAGE_FORMATS.some((format) =>
                 this.name.toLowerCase().endsWith(format)
             );
-            if (!isFileRenderableImage) {
-                return undefined;
+            if (isFileRenderableImage) {
+                return this.downloadPath;
+            }
+            if (this.cloudPath.endsWith(".zarr")) {
+                try {
+                    if (this.path?.startsWith("/allen")) {
+                        return await renderZarrThumbnailURL(
+                            `${AICS_FMS_FILES_NGINX_SERVER}${this.path}`
+                        );
+                    }
+                    return await renderZarrThumbnailURL(this.path);
+                } catch (error) {
+                    console.error("Error generating Zarr thumbnail:", error);
+                    throw new Error("Unable to generate Zarr thumbnail");
```

Comment:
I'm curious if we need both the console error and the throw here

Reply:
removed in [552170f](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/143/commits/552170f1bcfa6f438376984f8f4c54127cff0db5)[552170f](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/143/commits/552170f1bcfa6f438376984f8f4c54127cff0db5)[552170f](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/143/commits/552170f1bcfa6f438376984f8f4c54127cff0db5)

Reply:
I think these are a symptom of lots of iterations trying to read Zarr images and adding tries to bypass some of the errors

Reply:
Definitely think error handling is good move, just that having both throw & log seemed redundant

Reply:
What would we want to add in error handling here? Additional context that it comes from trying to render the image? The error is re-raised previously with the added console log so my thought was just let the error show if something was broken.